src/provider/store/Store.ts

```diff
@@ -17,15 +17,15 @@ export abstract class Action<S> {
         this.type = type;
     }
 
+    public async dispatch(store: StoreAPI<S, Action<S>>): Promise<void> {
```

Comment:
Do we need this? Could be persuaded on this way, but doing new Action().dispatch(store), at least right now, feels inside out compared to store.dispatch(new Action).

Edit: Ah, ok, I think I get this now - so we do store.dispatch(new Action()) for most Actions, but new Action().dispatch(store) for what were AsyncActions?